pytests/tests/test_pyclasses.py

```diff
@@ -82,9 +89,7 @@ def test_no_constructor_defined_propagates_cause(cls: Type):
             cls()  # should raise TypeError("No constructor defined for ...")
 
     assert exc_info.type is TypeError
-    assert exc_info.value.args == (
-        "No constructor defined for ClassWithoutConstructor",
-    )
+    assert exc_info.value.args == (exc_message,)
```

Comment:
This test will be run on all of Python 3.7 through 3.13, so I think the parameterized test is a good start but this will need to account for the variety between versions too.